sys/vm/vm_map.c

```diff
@@ -3337,9 +3337,8 @@ vm_map_protect(vm_map_t map, vm_offset_t start, vm_offset_t end,
 		 * potential compatibility breakage.  The alternative
 		 * would maximize compatibility.
 		 */
-		if (flags & VM_MAP_PROTECT_KEEP_CAP &&
-		    (((flags & VM_MAP_PROTECT_SET_MAXPROT) != 0 && (entry->max_protection & VM_PROT_CAP) != 0) ||
-		    ((flags & VM_MAP_PROTECT_SET_MAXPROT) == 0 && (entry->protection & VM_PROT_CAP) != 0))) {
+		if ((flags & VM_MAP_PROTECT_KEEP_CAP) != 0 &&
+		    (entry->max_protection & VM_PROT_CAP) != 0) {
```

Comment:
I don't think it affects correctness, but shouldn't we skip this block if `(flags & VM_MAP_PROTECT_SET_PROT) == 0`?

Reply:
I added an assert to accomplish this as it's a kernel bug to call with VM_MAP_PROTECT_KEEP_CAP and not at least one of VM_MAP_PROTECT_SET_PROT or VM_MAP_PROTECT_SET_MAXPROT